lib/services/sku-pack-service.js

```diff
@@ -271,5 +337,50 @@ function skuPackServiceFactory(
             });
     };
 
+    function getObjectsWithKey(obj, key) {
+        if (_.has(obj, key)) {
+            return [obj];
+        }
+        var res = [];
+        _.forEach(obj, function(v) {
+            if (typeof v === "object" && (v = getObjectsWithKey(v, key)).length) {
+                res.push.apply(res, v);
+            }
+        });
+
+        return res;
+    }
+
+    function loadWorkflowItems(skuName, fromRoot, dbCatalog) {
+        return fs.readdirAsync(fromRoot)
+            .filter(function(entry) {
+                return fs.statSync(fromRoot + '/' + entry).isFile();
+            })
+            .map(function(entry) {
+                return fs.readFileAsync(fromRoot + '/' + entry);
+            })
+            .filter(function(contents) {
+                var data = JSON.parse(contents);
+                return dbCatalog.findOne({injectableName: data.injectableName + '::' + skuName})
+                    .then(function(item) {
+                        return (typeof(item) === 'undefined') ? true : false;
```

Comment:
Isn't waterline going to return `null` here? Which will pass this check because `typeof(null) === 'object'`